### ISSUE TITLE
twisted.python.modules.PythonPath docstring issue

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4009]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)| @lvh reported|
|-|-|
|Trac ID|trac#4009|
|Type|defect|
|Created|2009-09-07 20:37:45Z|
|Branch|https://github.com/twisted/twisted/tree/pythonpath-docstr-4009|

Docstring documents a read only property (attribute) called `_sysPath`, although the thing is actually called `sysPath` (no underscore).

I'm making a (tiny ;-)) patch to fix it.

Attachments:

* [4009-docstring-typo.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e4a/e4a4a2ffad8ff5fed730a0790aad0700ed653fda/1733733425ac5c660949ef89a8be965ffc59135f.patch) (554 bytes) - added by lvh on 2009-09-07 20:39:49Z - smallest patch ever
* [pythonpath-doc-4009.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e4a/e4a4a2ffad8ff5fed730a0790aad0700ed653fda/05b3739f899ff254a29ccb4fd34cfca0e93d7158.patch) (1215 bytes) - added by thijs on 2013-01-22 21:56:43Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4009 4009
type__defect defect
reporter__lvh lvh
priority__lowest lowest
milestone__ 
branch__branches_pythonpath_docstr_4009 branches/pythonpath-docstr-4009
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1252355865000000 1252355865000000
changetime__1359107413000000 1359107413000000
version__None None
owner__thijs thijs
cc__glyph cc__therve cc__exarkun cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_2"></a>|@lvh commented|
|-|-|

For maximum reviewing ease, I'd like to point out the following line:

   605      sysPath = property(_getSysPath)

Look ma, no underscores :-)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

There should be documentation for both `_sysPath` and `sysPath`.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_4"></a>|@lvh commented|
|-|-|

I don't even get why `self._sysPath` is even set at all there. Nobody uses it, the public api is self.sysPath (a property which calls a factory -- if it is supplied in `__init__`, uses that, else the factory is just a zero-param lambda that returns the sysPath sequence passed to `__init__`).

I removed the line and the test suite passes. What gives? :-)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @glyph|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @lvh|
|-|-|

If you look at the documentation of sysPathFactory, you'll see the point: self._sysPath should be returned instead of calling sysPathFactory if provided. There should probably be a check if both are provided to raise an error.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Okay, so.  I think this ticket is describing something that's not true.  There's no typo here, the `@ivar` is documenting exactly what it intends to - the `_sysPath` attribute.  It's true that `_sysPath` is unused, though.  I can only guess that it's present to simplify debugging; otherwise it would only be available by digging into the closure that's created in `__init__`.

`sysPath` has documentation - in the docstring of the getter function for it.

So I think this should be closed as invalid.  However, since it seems that `sysPath` doesn't show up very reasonably in the pydoctor generated API docs, I think it would also be acceptable to change something here to improve that situation.  Perhaps that means adding an `@ivar` for `sysPath`?  I'm not sure if there's a better way than this to convey the necessary information to pydoctor.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [exarkun](#note_7):
> `sysPath` has documentation - in the docstring of the getter function for it.
> 
> Perhaps that means adding an `@ivar` for `sysPath`?

Attached patch uses the content of the docstring for the getter function of `sysPath` for the `@ivar` description. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set owner to @thijstriemstra<br>@exarkun set status to `new`|
|-|-|

Thanks.

  1. You missed a `sys.path` in the first sentence of the `PythonPath` class docstring
  1. `sysPath` should have a `@type`
  1. First letter of first word of `sysPath` doc should be capitalized.

Otherwise looks fine.  Please fix the above, add a misc news fragment, force a build to make sure there are no surprises on buildbot, and then merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra commented|
|-|-|

(In [36900]) Branching to 'pythonpath-docstr-4009'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra commented|
|-|-|

(In [36901]) address review comments, add news file. refs [#4009](https://github.com/twisted/twisted/issues/4009)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra commented|
|-|-|

Fixed the issues and forced a new [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/pythonpath-docstr-4009). All good except the `twistedcheckers` builder failed with [an error](https://buildbot.twistedmatrix.com/builders/twistedchecker/builds/136/steps/run-twistedchecker/logs/new%20twistedchecker%20errors) I'm not familiar with:

```
W9207:528,0:PythonPath: Missing a blank line before epytext markups
```

How can I resolve this, or can it be ignored?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun commented|
|-|-|

> How can I resolve this, or can it be ignored?

See https://bugs.launchpad.net/twistedchecker/+bug/1093502

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_17"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [exarkun](#note_16):
> > How can I resolve this, or can it be ignored?
> 
> See https://bugs.launchpad.net/twistedchecker/+bug/1093502

Thanks, I'll go ahead and merge then.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_18"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [36919]) Merge pythonpath-docstr-4009: Documented PythonPath.sysPath because it was missing from the API documentation.

Author: thijs
Reviewer: exarkun
Fixes: [#4009](https://github.com/twisted/twisted/issues/4009)
